source/MainPanel.cpp

```diff
@@ -106,6 +106,18 @@ void MainPanel::Step()
 		isActive = false;
 	}
 
+	// Offer the next available in flight mission.
+	if(isActive && player.HasAvailableEnteringMissions() && player.Flagship())
+	{
+		Mission *mission = player.EnteringMission();
+		if(mission && mission->HasSpace(*player.Flagship()))
+			mission->Do(Mission::OFFER, player, GetUI());
+		else
+			player.HandleBlockedEnteringMissions(GetUI());
```

Comment:
If I'm reading this right, we check only the first mission from the list, but if that fails, fire the blocked behavior of all available missions?

Reply:
EnteringMission will only return something that isn't nullptr for missions that the player already has space for (which does actually mean that the mission->HasSpace check here is redundant). This whole block of code only gets run if there are available entering missions that haven't been offered yet, though. That means that the only time that HandleBlockedEnteringMissions can run is if the only remaining missions are blocked.